src/Main.cpp

```diff
@@ -114,10 +116,11 @@ static void SurroundSelectionsWith(char ch1, char ch2) {
 
 		editor.ReplaceTarget(target);
 
+		// 2019-04-11:MVINCENT: leave cursor at END of insertion instead of beginning
 		if (i == 0)
-			editor.SetSelection(selection.first + offset + 1, selection.second + offset + 1);
+			editor.SetSelection(selection.second + offset + 1, selection.first + offset + 1);
```

Comment:
Any particular reason to swap them?

Reply:
These commands re-highlight the originally highlighted word after the surround text characters are added.  In the current version, they are highlighted and the blinking cursor is left at the beginning of the highlight.  I think it should be at the end of the highlighted word, closer to where someone would continue to type.

The call seems to be:  int SetSelection(int caret, int anchor)

So reversing the order puts the cursor (caret) at the "second" (read: further towards the end of the line) position and the anchor "first".

Reply:
When I first did this I put the caret at the beginning, because to me it felt more natural when wrapping the selection with multiple characters.

For example if you wanted to transform `text` into `{["text"]}` it felt natural for the cursor to stay at the beginning of the selection because you would type `{["` left to right. Not really a strong reason...just had to do something at the time. VS Code also does it the way you proposed. So probably best to be consistent with other editors since there's no *real* reason to do it my way :)